streaming/index.js

```diff
@@ -238,15 +257,34 @@ const startServer = async () => {
       // Write the HTTP response manually:
       socket.end(`HTTP/1.1 ${statusCode} ${http.STATUS_CODES[statusCode]}\r\n${Object.keys(headers).map((key) => `${key}: ${headers[key]}`).join('\r\n')}\r\n\r\n`);
 
+      // Finally, log the error:
+      request.log.error({
+        err,
+        res: {
+          statusCode,
+          headers
+        },
+        responseTime: Date.now() - startTime
+      }, err.message);
+
       return;
     }
 
     wss.handleUpgrade(request, socket, head, function done(ws) {
       // Remove the error handler:
       socket.removeListener('error', onSocketError);
 
+      request.log.info("Authenticated request & upgraded to WebSocket connection");
+
+      const wsLogger = logger.child({
+        req: {
+          id: request.id
+        },
+        accountId: resolvedAccount.accountId
+      });
+
       // Start the connection:
-      wss.emit('connection', ws, request);
+      wss.emit('connection', ws, request, wsLogger);
```

Comment:
For now I'm adding a third argument to `onConnection`, but in the future, we may be able to attach the "logger" directly to the `websocket` (`ws`)